### ISSUE TITLE
Extended config overriding inline rules

### ISSUE THREAD

**Issue by ChaosExAnima** [stylelint/stylelint#586]
Found an issue where a file loaded in the config `extends` block is actually taking precedence over the inline rules. Test case is as follows:

first.json:
```json
{
	"extends": [ "./second.json" ],
	"rules": {
		"string-quotes": "double"
	}
}
```

second.json:
```json
{
	"rules": {
		"string-quotes": "single"
	}
}
```
test.css:
```css
h1 {
	content: "x";
}
```
Result:
```cli
dist$ ./cli.js test.css

test.css
2:11	Expected single quotes (string-quotes)
```

**Comment by satazor** [comment]
I've experiencing the same issue, is this a regression?

**Comment by davidtheclark** [comment]
Released in 3.2.0
